vector-stores/spring-ai-elasticsearch-store/src/main/java/org/springframework/ai/elasticsearch/vectorstore/ElasticsearchVectorStore.java

```diff
@@ -297,4 +305,94 @@ private String getSimilarityMetric() {
 	public record ElasticSearchDocument(String id, String content, Map<String, Object> metadata, float[] embedding) {
 	}
 
+	/**
+	 * Creates a new builder instance for ElasticsearchVectorStore.
+	 * @param restClient the Elasticsearch REST client
+	 * @return a new ElasticsearchBuilder instance
+	 */
+	public static ElasticsearchBuilder builder(RestClient restClient) {
+		return new ElasticsearchBuilder(restClient);
+	}
+
+	public static class ElasticsearchBuilder extends AbstractVectorStoreBuilder<ElasticsearchBuilder> {
+
+		private final RestClient restClient;
+
+		private ElasticsearchVectorStoreOptions options = new ElasticsearchVectorStoreOptions();
+
+		private boolean initializeSchema = false;
+
+		private BatchingStrategy batchingStrategy = new TokenCountBatchingStrategy();
+
+		private FilterExpressionConverter filterExpressionConverter = new ElasticsearchAiSearchFilterExpressionConverter();
+
+		/**
+		 * Creates a new builder instance with the specified REST client.
+		 * @param restClient the Elasticsearch REST client
+		 * @throws IllegalArgumentException if restClient is null
+		 */
+		public ElasticsearchBuilder(RestClient restClient) {
+			Assert.notNull(restClient, "RestClient must not be null");
+			this.restClient = restClient;
+		}
+
+		/**
+		 * Sets the Elasticsearch vector store options.
+		 * @param options the vector store options to use
+		 * @return the builder instance
+		 * @throws IllegalArgumentException if options is null
+		 */
+		public ElasticsearchBuilder options(ElasticsearchVectorStoreOptions options) {
+			Assert.notNull(options, "options must not be null");
+			this.options = options;
+			return this;
+		}
+
+		/**
+		 * Sets whether to initialize the schema.
+		 * @param initializeSchema true to initialize schema, false otherwise
+		 * @return the builder instance
+		 */
+		public ElasticsearchBuilder initializeSchema(boolean initializeSchema) {
+			this.initializeSchema = initializeSchema;
+			return this;
+		}
+
+		/**
+		 * Sets the batching strategy for vector operations.
+		 * @param batchingStrategy the batching strategy to use
+		 * @return the builder instance
+		 * @throws IllegalArgumentException if batchingStrategy is null
```

Comment:
Where do we throw IllegalArgumentException? Same question for other places we specify that the exception is thrown for invalid values.

Reply:
In the method implementation, the `Assert.notNull(..)` call would throw the IAE.